core/use-swr.ts

```diff
@@ -521,6 +523,9 @@ export const useSWRHandler = <Data = any, Error = any>(
     // Keep the original key in the cache.
     setCache({ _k: fnArg })
 
+    // if a key is already active and has error, we should not trigger revalidation
+    if (isKeyHasActiveInstance && !isUndefined(error)) return
```

Comment:
This condition gets `unsubEvents` (L516) not to be called when unmounting. It would have to be put in the following condition .

```js
if (shouldDoInitialRevalidation)
```

https://github.com/vercel/swr/pull/2338/files#diff-134c45c7d0aef1e568828f8f2ee0f22398adc692b99cea901978c568ca0836caR530

Reply:
Thank you !